tiledb/api/src/error.rs

```diff
@@ -84,6 +86,64 @@ impl Display for DatatypeErrorKind {
     }
 }
 
+#[derive(Clone, Debug)]
+pub enum ObjectTypeErrorKind {
+    InvalidDiscriminant(u64),
+    TypeMismatch {
+        user_type: &'static str,
```

Comment:
In this case you could probably do `user_type: ObjectType`.
For datatype this field is something like `u32` or `f64` or what have you, so the mismatch is between the user programattic type and the TileDB data type, which isn't the same as the mismatch here

Reply:
Oh wait, yea sorry! Actually this invalid discriminant error is used in the C API conversion, since the CAPI stores this enum as ints under the hood.

Reply:
This comment is in reference to the `TypeMismatch` variant. I don't follow your comment.

Is `TypeMismatch` used anywhere? I checked out your branch and deleted it and it doesn't appear to be.  Better to remove it.  Looks the same for `ModeErrorKind`.

Reply:
Deleted the type mismatch errors.